simulator/simulator.js

```diff
@@ -199,8 +200,14 @@ prosumerBuffer = async function () {
                 let conserve = -diff * (1 - prosumer.under_production_buy)
 
                 if(prosumer.balance >= buy){
-                    await server_db_utils.updateBalanceById(prosumer.id, (prosumer.balance - buy).toFixed(2))
-                    await server_db_utils.updateBufferById(prosumer.id, (prosumer.buffer - conserve).toFixed(2))
+                    if(prosumer.buffer - conserve < 0){
+                        if(prosumer.buffer - conserve< 0){
```

Comment:
Oh fuck... I think we double stacked here somehow? I don't think I can edit

Reply:
haha shit, i fix!